.github/.check_scopes.lua

```diff
@@ -0,0 +1,38 @@
+function extractScopesFromFile(filename)
+    local scopes = {}
+    local file = io.open(filename, "r")
+    if not file then
+        error("Failed to open file: " .. filename)
+    end
+    for line in file:lines() do
+        local scope = line:match("%S+")
+        if scope then
+        scopes[scope] = true
+        end
+    end
+    file:close()
+    return scopes
+end
+
+function extractScopeFromCommit(commitTitle)
+    local scope = commitTitle:match("%a+%((.-)%)")
+    return scope
+end
+  
+local filename = ".github/.scopes.txt"
+local scopes = extractScopesFromFile(filename)
+  
+-- Get commit titles using git log command
+local handle = io.popen('git log --pretty=format:"%s" $(git merge-base HEAD origin/main)..HEAD')
+local commits = handle:read("*a")
+handle:close()
+
+-- Check if the scope matches for each commit
+for line in commits:gmatch("[^\r\n]+") do
+  local commitScope = extractScopeFromCommit(line)
+  if commitScope and not scopes[commitScope] then
+    print("Invalid scope:", commitScope)
+    print("Commit:", line)
+    os.exit(1)
+  end
+end
```

Comment:
Inconsistent indentation.
This part uses 2 spaces, while the section above uses 4 spaces.